fs/accounting/accounting.go

```diff
@@ -14,23 +16,87 @@ import (
 	"github.com/pkg/errors"
 )
 
+type jobCtx int64
+
+const jobKey jobCtx = 1
+
+// WithJobID returns copy of the parent with assigned jobID.
+func WithJobID(parent context.Context, jobID int64) context.Context {
+	return context.WithValue(parent, jobKey, jobID)
+}
+
+// JobIDFromContext returns job ID from the context if it's available.
+func JobIDFromContext(ctx context.Context) (int64, bool) {
+	jobID, ok := ctx.Value(jobKey).(int64)
+	return jobID, ok
+}
+
 // ErrorMaxTransferLimitReached is returned from Read when the max
 // transfer limit is reached.
 var ErrorMaxTransferLimitReached = fserrors.FatalError(errors.New("Max transfer limit reached as set by --max-transfer"))
 
+// AccountSnapshot represents state of an account at point in time.
+type AccountSnapshot struct {
+	Name      string `json:"name"`
+	Size      int64  `json:"size"`
+	Bytes     int64  `json:"bytes"`
+	Checked   bool   `json:"checked"`
+	Timestamp msTime `json:"timestamp"`
+	Error     error  `json:"-"`
+	JobID     int64  `json:"jobid"`
+}
+
+// MarshalJSON implements json.Marshaler interface.
+func (as AccountSnapshot) MarshalJSON() ([]byte, error) {
+	err := ""
+	if as.Error != nil {
+		err = as.Error.Error()
+	}
+	type Alias AccountSnapshot
+	return json.Marshal(&struct {
+		Error string `json:"error"`
+		Alias
+	}{
+		Error: err,
+		Alias: (Alias)(as),
+	})
+}
+
+// NewCheckingSnapshot creates snapshot for checked object.
+func NewCheckingSnapshot(obj fs.Object) AccountSnapshot {
+	return AccountSnapshot{
+		Name:      obj.Remote(),
+		Size:      obj.Size(),
+		Bytes:     obj.Size(),
```

Comment:
It would be better if `"bytes": total transferred bytes for this file,` would be the actually transferred bytes. If we have a transferred entry we can assume success if error is "".